Nextflow pipelines
==================

We offer a range of sequencing processing pipelines. We develop them using `Nextflow <https://www.nextflow.io/>`_ workflow language.

For the end-users we offer Nextflow pipelines available to run on both Sanger *LSF* farm or *OpenStack* flexible compute environment. The pipelines can be run either by us or by the end-users themselves.

Here is the list of pipelines we develop:

1. `scRNAseq (10X cellranger) <https://github.com/cellgeni/10xcellranger>`_
2. `RNA-seq (for Smartseq2 and bulk data) <https://github.com/cellgeni/rnaseq-noqc>`_
3. `souporcell (for 10X data) <https://github.com/wheaton5/souporcell>`_
4. `ATAC-seq (for 10X and Smartseq2 data) <https://github.com/cellgeni/cellatac>`_
5. `TraCeR/BraCer (for Smartseq2 data) <https://github.com/cellgeni/tracer>`_
6. `scHLAcount (for 10X data) <https://github.com/10XGenomics/scHLAcount>`_
7. `cellSNP (for 10X data) <https://github.com/single-cell-genetics/cellSNP>`_
8. `velocyto (for 10X data) <http://velocyto.org/velocyto.py/tutorial/index.html#running-the-cli>`_
9. `STARsolo (for 10X data) <https://github.com/alexdobin/STAR>`_
10. `Data Sharing <https://github.com/cellgeni/guitar>`_

.. note:: If you would like to run ``cellranger`` by *10X* please put in a ticket for the NPG group by emailing to new-seq-pipe@sanger.ac.uk and specify which version of the cellranger and the genome you would like to use.

