

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Services we provide &mdash; Cellular Genetics IT 0.2 documentation</title>
  

  
  
  
  

  

  
  
    

  

  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Useful information" href="utils.html" />
    <link rel="prev" title="Our vision" href="approach.html" /> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Cellular Genetics IT
          

          
          </a>

          
            
            
              <div class="version">
                0.2
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="history.html">Our story</a></li>
<li class="toctree-l1"><a class="reference internal" href="remit.html">Our remit</a></li>
<li class="toctree-l1"><a class="reference internal" href="approach.html">Our vision</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Services we provide</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#ticketing-system">Ticketing system</a></li>
<li class="toctree-l2"><a class="reference internal" href="#interactive-analysis-environment">Interactive analysis environment</a></li>
<li class="toctree-l2"><a class="reference internal" href="#run-a-pipeline">Run a pipeline</a></li>
<li class="toctree-l2"><a class="reference internal" href="#share-data-with-external-collaborators">Share data with external collaborators</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#globus">Globus</a></li>
<li class="toctree-l3"><a class="reference internal" href="#cram-files">cram files</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#data-submission">Data submission</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="utils.html">Useful information</a></li>
<li class="toctree-l1"><a class="reference internal" href="mlwarehouse.html">ML Warehouse</a></li>
<li class="toctree-l1"><a class="reference internal" href="contact.html">How to contact us</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Cellular Genetics IT</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Services we provide</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/services.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="services-we-provide">
<h1>Services we provide<a class="headerlink" href="#services-we-provide" title="Permalink to this headline">¶</a></h1>
<div class="section" id="ticketing-system">
<h2>Ticketing system<a class="headerlink" href="#ticketing-system" title="Permalink to this headline">¶</a></h2>
<p>We operate using JIRA ticketing system. To be able to put in tickets you need to contact us by either emailing to <a class="reference external" href="mailto:vk6&#37;&#52;&#48;sanger&#46;ac&#46;uk">vk6<span>&#64;</span>sanger<span>&#46;</span>ac<span>&#46;</span>uk</a> or using <a class="reference external" href="https://mattermost.sanger.ac.uk/cellgeninf/channels/jira-requests">our Mattermost channel</a> (please use your Sanger credentials to login).</p>
<p>Once registered you can put in a ticket using either <a class="reference external" href="https://cellgeni.atlassian.net/servicedesk/customer/portal/1">our portal</a> or by emailing to <a class="reference external" href="mailto:support&#37;&#52;&#48;cellgeni&#46;atlassian&#46;net">support<span>&#64;</span>cellgeni<span>&#46;</span>atlassian<span>&#46;</span>net</a>.</p>
</div>
<div class="section" id="interactive-analysis-environment">
<h2>Interactive analysis environment<a class="headerlink" href="#interactive-analysis-environment" title="Permalink to this headline">¶</a></h2>
<p>We support a Jupyter Hub server running on Sanger Cloud. It is available for everyone in our Programme. Jupyter allows you to run your analysis in multiple environments (<code class="docutils literal notranslate"><span class="pre">R</span></code>, <code class="docutils literal notranslate"><span class="pre">python</span></code>, <code class="docutils literal notranslate"><span class="pre">Julia</span></code>, etc.) and also to create and share notebooks containing your analysis, code, equations and visualisations. We think this is an ideal environment for any kind of downstream analysis. For more details please refer to <a class="reference external" href="http://jupyter.org/hub">Jupyterhub documentation</a></p>
<p>Before logging in to JupyterHub please read our instructions here:
<a class="reference external" href="https://github.com/cellgeni/notebooks#user-instructions">https://github.com/cellgeni/notebooks#user-instructions</a></p>
<p>Once you read the instructions go ahead and login with you Sanger credentials here (it may take some time when you do it first time):
<a class="reference external" href="https://jupyter.cellgeni.sanger.ac.uk">https://jupyter.cellgeni.sanger.ac.uk</a></p>
<p>We provide some notebook templates with the pre-installed software. These are located in the <code class="docutils literal notranslate"><span class="pre">notebooks</span></code> folder. Corresponding example data is located in the <code class="docutils literal notranslate"><span class="pre">data</span></code> folder. Before running your analysis, please make a copy of a notebook template, save it to your home folder and work with the copy. At the moment we have the following notebook templates:</p>
<ol class="arabic simple">
<li><a class="reference external" href="https://github.com/cellgeni/notebooks/blob/master/files/notebooks/10X-scanpy.ipynb">scanpy notebook for analysis of 10X data</a></li>
<li><cite>Seurat notebook for analysis of 10X data &lt;https://github.com/cellgeni/notebooks/blob/master/files/notebooks/10X-Seurat.Rmd&gt;</cite></li>
</ol>
</div>
<div class="section" id="run-a-pipeline">
<h2>Run a pipeline<a class="headerlink" href="#run-a-pipeline" title="Permalink to this headline">¶</a></h2>
<p>We offer a range of sequencing processing pipelines. We develop them in two workflow languages: <a class="reference external" href="https://www.nextflow.io/">Nextflow</a> (very user-friendly) and <a class="reference external" href="https://www.commonwl.org/">CWL</a> (universal standard).</p>
<p>For the end-users we offer Nextflow pipelines available to run on both Sanger <em>LSF</em> farm or <em>OpenStack</em> flexible compute environment. The pipelines can be run either by us or by the end-users themselves.</p>
<p>Here is the list of pipelines we develop:</p>
<ol class="arabic simple">
<li><a class="reference external" href="https://github.com/cellgeni/rnaseq-noqc">RNA-seq</a> (both bulk and single cell)</li>
<li><a class="reference external" href="https://github.com/cellgeni/chipseq">Chip-seq</a></li>
<li><a class="reference external" href="https://github.com/cellgeni/atacseq">ATAC-seq</a></li>
</ol>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If you would like to run <code class="docutils literal notranslate"><span class="pre">cellranger</span></code> by <em>10X</em> please put in a ticket for the NPG group by emailing to <a class="reference external" href="mailto:new-seq-pipe&#37;&#52;&#48;sanger&#46;ac&#46;uk">new-seq-pipe<span>&#64;</span>sanger<span>&#46;</span>ac<span>&#46;</span>uk</a> and specify which version of the cellranger and the genome you would like to use.</p>
</div>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">Our pipelines start from aligned cram files and mostly concentrate on the downstream analysis (starting from the count matrix). Therefore, when submitting your experiment to sequencing please make sure you specify the correct aligner for your data.</p>
</div>
</div>
<div class="section" id="share-data-with-external-collaborators">
<h2>Share data with external collaborators<a class="headerlink" href="#share-data-with-external-collaborators" title="Permalink to this headline">¶</a></h2>
<div class="section" id="globus">
<h3>Globus<a class="headerlink" href="#globus" title="Permalink to this headline">¶</a></h3>
<p>We use <a class="reference external" href="https://www.globus.org/">Globus</a> network to share the data with external collaborators. It allows us to share data from e.g. a specific folder on the Sanger LFS cluster directly with the external world.</p>
<p>The sharing process consists of the following steps:</p>
<ol class="arabic simple">
<li>We share the data with the user’s personal/work email address</li>
<li>The user <a class="reference external" href="https://docs.globus.org/how-to/get-started/">creates/logs in their Globus account</a> using the sharing email</li>
<li>The user needs to create a personal Globus endpoint either on their <a class="reference external" href="https://docs.globus.org/how-to/globus-connect-personal-linux/">Linux laptop / compute cluster</a> or on their <a class="reference external" href="https://docs.globus.org/how-to/globus-connect-personal-mac/">Mac laptop</a> or on their <a class="reference external" href="https://docs.globus.org/how-to/globus-connect-personal-windows/">Windows laptop</a>.</li>
<li>The user activates their personal Globus endpoint by starting globus from the command line if on a cluster/Linux, or by starting the globus application if on Mac/Windows.</li>
<li>Once the users personal endpoint is setup they can transfer the data by simply <a class="reference external" href="https://docs.globus.org/how-to/get-started/">logging in to their Globus account using the sharing email address and drag and dropping the data</a>.</li>
</ol>
<p>For more information please visit the <a class="reference external" href="https://docs.globus.org/how-to/">Globus official documentation</a>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If the user would like to check MD5 hash, the MD5 sum files will be located in the same sharing folder with the data files.</p>
</div>
</div>
<div class="section" id="cram-files">
<h3>cram files<a class="headerlink" href="#cram-files" title="Permalink to this headline">¶</a></h3>
<p>Sanger default file format for storing NGS data is <code class="docutils literal notranslate"><span class="pre">CRAM</span></code> and this is what we provide to the user when share data with them. Typically <code class="docutils literal notranslate"><span class="pre">CRAM</span></code> achieves 40-50% space saving over the alternative <code class="docutils literal notranslate"><span class="pre">BAM</span></code> format and much more than that over the compressed <code class="docutils literal notranslate"><span class="pre">fastq</span></code> files. For more information please visit <a class="reference external" href="https://www.sanger.ac.uk/science/tools/cram">this page</a>.</p>
<p>Once the user obtained the data from Globus, the data can be converted from <code class="docutils literal notranslate"><span class="pre">CRAM</span></code> to <code class="docutils literal notranslate"><span class="pre">fastq</span></code> format using the following steps:</p>
<ul class="simple">
<li>Install <code class="docutils literal notranslate"><span class="pre">samtools</span></code> with version <strong>&gt;=1.8</strong> (in this case <code class="docutils literal notranslate"><span class="pre">samtools</span></code> should automatically download the right genome reference if your local installation does not have it)</li>
<li>Run the following commands (set NCPU to a number of CPUs, if you are on a multi-cpu machine). This will create paired fastq files <code class="docutils literal notranslate"><span class="pre">samplename_1.fastq.gz</span></code> and <code class="docutils literal notranslate"><span class="pre">samplename_2.fastq.gz</span></code>:</li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>samtools collate -O -u -@ NCPU samplename.cram tmppfx <span class="p">|</span> <span class="se">\</span>
    samtools fastq -N -F 0x900 -@ NCPU -1 samplename_1.fastq.gz -2 samplename_2.fastq.gz -
</pre></div>
</div>
<p>If this does not work, you could try running these first:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">unset</span> REF_PATH
<span class="nb">unset</span> REF_CACHE
</pre></div>
</div>
</div>
</div>
<div class="section" id="data-submission">
<h2>Data submission<a class="headerlink" href="#data-submission" title="Permalink to this headline">¶</a></h2>
<p>To submit your data to a public repository, e.g. <a class="reference external" href="https://www.ebi.ac.uk/ega/home">EGA</a> or <a class="reference external" href="https://www.ebi.ac.uk/ena">ENA</a>, please contact <a class="reference external" href="mailto:datahose&#37;&#52;&#48;sanger&#46;ac&#46;uk">datahose<span>&#64;</span>sanger<span>&#46;</span>ac<span>&#46;</span>uk</a>. More information can be found <a class="reference external" href="https://stackoverflow.sanger.ac.uk/question/935801792929730560/how-to-submit-data-to-a-public-repository">here</a>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="utils.html" class="btn btn-neutral float-right" title="Useful information" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="approach.html" class="btn btn-neutral" title="Our vision" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018, Cellular Genetics IT.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'0.2',
            LANGUAGE:'None',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>